Add support for all types of monthly repeating schedules

bcantoni opened the pull request:
This should fix handling all of the "monthly" schedule types now; see details in #1358 and #1369.

bcantoni commented:
@jacalata and/or @jorwoods if you could help review that would be great.

anyoung-tableau commented on tableauserverclient/models/interval_item.py:
```diff
@@ -246,21 +246,35 @@ def interval(self):
-                        raise ValueError(error)
+            try:
+                if not (1 <= int(interval_value) <= 31):
+                    error = f"Invalid monthly numeric frequency interval: {interval_value}."
```
This error message can never happen right? Its ValueError will get caught below and either swallowed or a new ValueError will be thrown. I haven't used Python much, is it common to use exceptions to control flow?

Maybe use the `isinstance(interval, (int, float))` construct like on line 80 to detect if it's a number first.

bcantoni replied:
Well in theory I think it would fail right there if the numerical value was 32 or "32".

I agree using exceptions here is a little funky. I was trying to fit in with the methods used already.

jorwoods replied:
Using exceptions for control flow isn't uncommon in python. The `except ValueError` on 274 would indeed catch and consume the one raised on 273. The type check proposed by @anyoung-tableau does seem like a good alternative to let the different errors surface.

bcantoni replied:
I'll take another look at it. The challenge is that even numbers are strings by the time they land here.

jacalata replied:
Easiest way to do it seems like explicitly listing the known allowed digits: 
for value in range(1, 32):
    VALID_INTERVALS.add(str(value))

bcantoni commented:
@jacalata I modified the validation and simplified by including the 1..31 range as both int or string. Looks better now and avoids that "never reachable" error path I started with.

jacalata reviewed (approved):
Looks good
